### PR TITLE
fix(context): Create a new {Api,System}Context on new provider

### PR DESCRIPTION
**Pull request by amaurymartiny**
Before: because we were using `useRef`, if we changed provider, it didn't create a new Api object (similar behavior with System).

After: we put api obect in state, and an effect to update state iif provider changed.

### CODE REVIEW

File: front/context/src/ApiContext.tsx [hunk 1]
```diff
@@ -30,25 +27,32 @@ export function ApiContextProvider(
...
-    api.isReady.subscribe(() => {
+    const subscription = api.isReady.subscribe(() => {
       l.log(`Api ready, app is ready to use`);
 
-      setState({ isApiReady: true });
+      setIsReady(true);
     });
-  }, [api, provider]);
+
+    return subscription.unsubscribe();
```
Comment (yjkimjunior) [inline front/context/src/ApiContext.tsx]:
```suggestion
    return () => subscription.unsubscribe();
```

Comment (amaurymartiny) [reply front/context/src/ApiContext.tsx]: Ah yeah

File: front/context/src/ApiContext.tsx [hunk 2]
```diff
@@ -30,25 +27,32 @@ export function ApiContextProvider(
...
+    const subscription = api.isReady.subscribe(() => {
       l.log(`Api ready, app is ready to use`);
 
-      setState({ isApiReady: true });
+      setIsReady(true);
     });
-  }, [api, provider]);
+
+    return () => subscription.unsubscribe();
+  });
```
Comment (yjkimjunior) [inline front/context/src/ApiContext.tsx]:
```suggestion
  }, [api]);
```
^ i would think this to be necessary

**Comment by amaurymartiny** [comment]
Yep, you're right, fixed

**Pull request merged**